src/layouts/Sidebar.tsx

```diff
@@ -13,6 +19,12 @@ function Sidebar(this: any) {
     setOpen(false);
   };
 
+  const conditionalProps = {
+    ...(userData?.email
+      ? { onClick: handleClickOpen }
+      : { component: Link, to: "/auth" }),
+  };
+
```

Comment:
To make it more readable it is better to move ternary outside of object literal:
```
const conditionalProps = userData?.email ? { onClick: handleClickOpen } : { component: Link, to: "/auth" };
```